### PR TITLE
Generate gRPC Method Descriptors

### PR DESCRIPTION
**Pull request by andrewparmet**
Generates service and method descriptors:

```kotlin
object OperationsGrpc {
    const val SERVICE_NAME = "google.longrunning.Operations"

    val serviceDescriptor: ServiceDescriptor by lazy {
        ServiceDescriptor.newBuilder(SERVICE_NAME)
            .addMethod(listOperationsMethod)
            .build()
    }

    val listOperationsMethod: MethodDescriptor<ListOperationsRequest, ListOperationsResponse> by lazy {
        MethodDescriptor.newBuilder<ListOperationsRequest, ListOperationsResponse>()
            .setType(MethodType.UNARY)
            .setFullMethodName(generateFullMethodName(SERVICE_NAME, "ListOperations"))
            .setRequestMarshaller(KtMarshaller(ListOperationsRequest))
            .setResponseMarshaller(KtMarshaller(ListOperationsResponse))
            .build()
    }
}
```

### CODE REVIEW

File: protokt-codegen/src/main/resources/services.stg [hunk 1]
```diff
@@ -13,30 +13,60 @@
...
 
->>
+    <methods : { m | val <m.lowerName>Method: MethodDescriptor\<<m.in>, <m.out>\> by lazy {
+    <m.body><\n><\\>
+    \}}; separator="\n\n">
 
-descriptorSupplier(fqcn) ::= <<
+    @JvmName("descriptor")
+    @JvmSynthetic
+    fun getServiceDescriptor() =
```
Comment (ogolberg) [inline protokt-codegen/src/main/resources/services.stg:32]: would `@get:JvmName` work?

Comment (andrewparmet) [reply protokt-codegen/src/main/resources/services.stg:32]: `This annotation is not applicable to target 'member function' and use site target '@get'`

Comment (andrewparmet) [reply protokt-codegen/src/main/resources/services.stg:32]: Strictly speaking the `@JvmSynthetic` annotation isn't need for this to compile, it's just to hide it so the getter is the only visible item from Java.

Comment (ogolberg) [reply protokt-codegen/src/main/resources/services.stg:32]: I meant `@get:JvmName`  on the `val`

Comment (andrewparmet) [reply protokt-codegen/src/main/resources/services.stg:32]: Ah - the catch is grpc-kotlin calls the `get()` functions _from Kotlin_, so it has to be present as a function in Kotlin.

Comment (andrewparmet) [reply protokt-codegen/src/main/resources/services.stg:32]: https://github.com/grpc/grpc-kotlin/blob/master/compiler/src/test/java/io/grpc/kotlin/generator/HelloWorldProtoGrpcKt.expected#L57

Comment (andrewparmet) [reply protokt-codegen/src/main/resources/services.stg:32]: So yes - `@get:JvmName` on the `val`s works, but then you have an extra visible member from Java. I don't know whether we care.

**Pull request merged**